test/dummy/app/models/setting.rb

```diff
@@ -2,6 +2,7 @@
 
 class Setting < RailsSettings::Base
   cache_prefix { "v1" }
+  permit_value_classes(ActiveSupport::Duration, Symbol, BigDecimal)
```

Comment:
If this is requirement, we can add in Base model as default for includes all possible classes.

Reply:
So, we don't need public this API for users.

Reply:
> If this is requirement

According to my description, it is a requirement for Ruby 3.1 as it upgrades Psych and the behavior of YAML.load changes through Psych.

> we can add in Base model as default for includes all possible classes.

What would "all possible classes" be?

------

I could add `ActiveSupport::Duration` in the Base model and that would work. However, it would only work for my case. Another user of the gem might not need this class or a third user of the gem might need another class that the Gem does not support just yet. They would need to open a new PR to support their case.


If we were to have this public API, users that have special cases like I have (`ActiveSupport::Duration, BigDecimal`) will need to configure it specifically for their needs.